backend/api/messenger/messenger.service.go

```
@@ -0,0 +1,111 @@
+	To               string           `json:"to"`
+	Type             string           `json:"type"`
+	Template         whatsappTemplate `json:"template"`
+}
+
+type whatsappTemplate struct {
+	Name       string                   `json:"name"`
+	Language   map[string]string        `json:"language"`
+	Components []map[string]interface{} `json:"components"`
+}
+
+func (s *messengerService) SendOTP(dto *types.SendOTPInputDTO) error {
+	url := fmt.Sprintf("%s/%s/%s/messages", FACEBOOK_GRAPH_API_BASE_URL, s.config.APIVersion, s.config.PhoneNumberID)
+
+	payload := whatsappMessage{
+		MessagingProduct: WHATSAPP_KEY,
+		To:               dto.Phone,
+		Type:             TEMPLATE_KEY,
+		Template: whatsappTemplate{
+			Name: types.GetTemplateName(dto.LanguageCode),
+			Language: map[string]string{
+				"code": dto.LanguageCode.String(),
+			},
+			Components: []map[string]interface{}{
+				{
+					"type": "body",
+					"parameters": []map[string]interface{}{
+						{
+							"type": "text",
+							"text": dto.OTP,
+						},
+					},
+				},
+				{
+					"type":     "button",
+					"sub_type": "url",
+					"index":    0,
+					"parameters": []map[string]interface{}{
+						{
+							"type": "text",
+							"text": dto.OTP,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	body, err := json.Marshal(payload)
+	if err != nil {
+		return fmt.Errorf("failed to marshal payload: %w", err)
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("Authorization", "Bearer "+s.config.AccessToken)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := s.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		respBody, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("whatsapp API error [%d]: %s", resp.StatusCode, string(respBody))
```

Does this error is returned to Customer? If yes, remove it from customer view, customer mustnt know about internal errors with integrations. Also is there any loggers catching this error?

---

This error is not returned to customer, since we only return the errors to handler layer, and it decides what output to show to the client, avoiding direct error details.